packages/react-dom/src/events/plugins/EnterLeaveEventPlugin.js

```diff
@@ -147,16 +148,17 @@ function extractEvents(
   leave.target = fromNode;
   leave.relatedTarget = toNode;
 
-  let enter = new SyntheticEvent(
+  let enter: KnownReactSyntheticEvent | null = new SyntheticEvent(
     enterEventType,
     eventTypePrefix + 'enter',
     to,
     nativeEvent,
     nativeEventTarget,
     eventInterface,
   );
-  enter.target = toNode;
-  enter.relatedTarget = fromNode;
+  // flow thinks `enter` could be `null` at this point
```

Comment:
Why

Reply:
We have to declare it as ` let enter: KnownReactSyntheticEvent | null` because we later re-assign it to `null` conditionally.

Reply:
I'm not familiar with flow so there might be a better alternative. The current statements seems sound to me. [TypeScript is fine with it](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyiB2wAWFgEsDGBRAbhRUAvFAN5QBOEAhhsAHJUC2EAXFAM7DlrwDmUAXwBQQgDaoo+YBHJs4iFOmx4CAHyjwArqNFFSlGvSasoAclMCA3EKkyAdAdoNme89aFoAZlAAUXTRAAlKRCUJKIMnpaOtYCQA), but [flow is not](https://flow.org/try/#0C4TwDgpgBAyiB2wAWFgEsDGBRAbhRUAvFAN5QBOEAhhsAHJUC2EAXFAM7DlrwDmUAXwBQQgDaoo+YBHJs4iFOmx4CAHyjwArqNFFSlGvSasoAclMCA3EKkyAdAdoNme89aFoAZlAAUXTRAAlKRCUJKIMnpaOtYCQA)

Reply:
You could have two variables. First one is not nullable and that’s where you initialise. Then you copy it to a second nullable one.

Reply:
Took the liberty and flipped the assignment around. Instead of conditionally nulling we conditionally create the event. This means no more wasted work in the constructor.

Otherwise we'll end up with 

```js
let enter: SyntheticEvent | null = null;
const enterEvent: SyntheticEvent = { reactName: 'mouseleave' }
enterEvent.reactName = '';
enter = enterEvent
```
which looks odd. Or do you prefer that?